src/SharpCompress/Factories/TarFactory.cs

```diff
@@ -0,0 +1,173 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+using SharpCompress.Archives;
+using SharpCompress.Archives.Tar;
+using SharpCompress.Archives.Zip;
+using SharpCompress.Common;
+using SharpCompress.Compressors;
+using SharpCompress.Compressors.BZip2;
+using SharpCompress.Compressors.LZMA;
+using SharpCompress.Compressors.Xz;
+using SharpCompress.IO;
+using SharpCompress.Readers;
+using SharpCompress.Readers.Tar;
+using SharpCompress.Writers;
+using SharpCompress.Writers.Tar;
+
+namespace SharpCompress.Factories
+{
+    /// <summary>
+    /// Represents the foundation factory of TAR archive.
+    /// </summary>
+    public class TarFactory : Factory,
+        IArchiveFactory,
+        IMultiArchiveFactory,
+        IReaderFactory,
+        IWriterFactory,
+        IWriteableArchiveFactory
+    {
+        #region IFactory
+
+        /// <inheritdoc/>
+        public override string Name => "Tar";
+
+        /// <inheritdoc/>
+        public override ArchiveType? KnownArchiveType => ArchiveType.Tar;
+
+        /// <inheritdoc/>
+        public override IEnumerable<string> GetSupportedExtensions()
+        {
+            yield return "tar";
+            yield return "tgz";
```

Comment:
more extensions for LZip/BZip2 types?  tbz or tlz?  those are guesses.

Reply:
I've just added the extensions reported by wikipedia, didn't know tar had so many flavours!

I was tempted to add the "tar.*" variants, but I am aware some APIs don't handle well extensions with dots and can produce misleading results.

Reply:
nah, we'll let the user handle nested extensions...

some of these we don't handle though... LZMA we could but XZ, zstd we don't *yet and not sure what compress is